views/default/navigation/pagination.php

```diff
@@ -41,6 +44,21 @@
 	$base_url = current_page_url();
 }
 
+if ($url_fragment && !preg_match('~#.~', $base_url)) {
+	// we'll sometimes add fragment
+	$get_href = function ($offset) use ($base_url, $offset_key, $url_fragment) {
```

Comment:
Why use anonymous functions? Can't we just add it somewhere more useful so that we can recycle if needed?